databases/week3/lesson3.sql

```diff
@@ -0,0 +1,144 @@
+-- Get a reservation with any id
+SELECT * FROM reservation WHERE id=1;
+-- Update a reservation with any id
+UPDATE reservation 
+SET number_of_guests = 5
+WHERE id=1;
+-- Delete a reservation with any id
+DELETE FROM reservation
+WHERE id = 1;
+-- Get all reviews
+SELECT * FROM review;
+-- Add a new review
+INSERT INTO review(`title`, `description`, `meal_id`, `stars`, `created_date`)
+VALUES('warm and welcoming', 'Cinzia gave a great explanation of the different foods, great instructions and was also warm and welcoming from her kitchen.', 4, 5, '2021-10-10');
+INSERT INTO review(`title`, `description`, `meal_id`, `stars`, `created_date`)
+VALUES('very flexible', 'Cinzia helped guide me and my friends on a very fun, very flexible, and very delicious video cooking adventure!', 3, 4, '2021-11-02');
+INSERT INTO review(`title`, `description`, `meal_id`, `stars`, `created_date`)
+VALUES('lovely', 'We felt like in Paris, cooking all together and chatting. Lovely!', 7, 5, '2021-10-30');
+INSERT INTO review(`title`, `description`, `meal_id`, `stars`, `created_date`)
+VALUES('highly recommend', ' Highly recommend her, and we look forward to cooking with her!', 2, 4, '2021-11-05');
+INSERT INTO review(`title`, `description`, `meal_id`, `stars`, `created_date`)
+VALUES('Slow service!', 'Nice location by the clay pit park and nice food, but it took 50 minutes from we ordered the food until it arrived', 2, 2, '2021-11-08');
+-- Get a review with any id
+SELECT * FROM review WHERE id=2;
+-- Update a review with any id
+UPDATE review 
+SET meal_id = 2
+WHERE id=2;
+-- Delete a review with any id
+DELETE FROM review
+WHERE id = 1;
+-- Get meals that has a price smaller than a specific price
+SELECT * FROM meal
+WHERE price > 250.00;
+-- Get meals that still has available reservations
+-- use of IS NULL to find also the meals that no reservations where made for, so they are fully available
+SELECT meal.id, meal.title, meal.max_reservations, SUM(reservation.number_of_guests) as reserved
+FROM meal
+LEFT JOIN reservation 
+ON reservation.meal_id = meal.id
+WHERE (max_reservations - number_of_guests > 0) OR number_of_guests IS NULL
+GROUP BY id;
+-- Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+SELECT * FROM meal
+WHERE title LIKE '%le%';
+-- Get meals that has been created between two dates
+SELECT * FROM meal
+WHERE month(created_date) BETWEEN 07 AND 10;
+-- Get only specific number of meals fx return only 5 meals
+SELECT * FROM meal
+ORDER BY id LIMIT 2, 5;
+-- Get the meals that have good reviews
+SELECT review.*, meal.title as meal_type FROM review
+LEFT JOIN meal ON meal_id = meal.id
+WHERE stars >= 4
+ORDER BY stars DESC;
+-- Get reservations for a specific meal sorted by created_date
+SELECT reservation.*, meal.title, meal.max_reservations FROM reservation 
+LEFT JOIN meal ON meal_id = meal.id
+WHERE meal_id=6
```

Comment:
Pls remove this line and try to modify it so that you can get the desired results
hint: ```
SELECT meal.Title, reservation.meal_Id, SUM(reservation.Number_of_guests), reservation.Created_date AS Res_created_date
FROM reservation
   JOIN meal ON (meal.id=reservation.meal_id)
GROUP BY reservation.created_date,meal.Title
ORDER BY reservation.created_date;
```

Reply:
I tried to modify and implement your suggestions. Thank you.